docs/book/src/migration/v2vsv3.md

```diff
@@ -2,21 +2,71 @@
 
 This document covers all breaking changes when migrating from v2 to v3.
 
-The details of all changes (breaking or otherwise) can be found in [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases) release notes.
+The details of all changes (breaking or otherwise) can be found in
+[controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/releases),
+[controller-tools](https://github.com/kubernetes-sigs/controller-tools/releases)
+and [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases)
+release notes.
+
+## Common changes
+V3 project uses go modules and the go version 1.15 wheredep is no longer supported.
 
 ## Kubebuilder
 
 - A plugin design was introduced to the project. For more info see the [Extensible CLI and Scaffolding Plugins][plugins-phase1-design-doc]
+
 - The GO supported version was upgraded from 1.13+ to 1.15+
 
-## Project config versions
+- New PROJECT file layout which stores the data used to do the scaffolds with the purpose to support the new features and allow future the CLI provide helpers to it users in the future. 
+
+## New plugin go/v3
+
+<aside class="note warning">
+<h1>Note</h1>
+
+From now on, new projects built with the tool will be using by default the go plugin `go.kubebuilder.io/v3+` which has the support for the new features and APIs.
+</aside>
+
+- Support for `apiextensions.k8s.io/v1` to replace `apiextensions.k8s.io/v1beta1` which was deprecated in Kubernetes version `1.16`.
```

Comment:
Could group these under a list item to make this a bit easier to read, like:

```suggestion
- Scaffolded/Generated API version changes
  * Use `apiextensions/v1` for generated CRDs (`apiextensions/v1beta1` was deprecated in Kubernetes 1.16)
  * Use `admissionregstration...` for generated webhooks (...)
  * ...
- Code changes
  * [manager flags]
  * [healthz]
  * [componentconfig]
 - Misc
   * Support for CT v0.4.1 & CR v0.8.0
   * Required envtest binaries are automatically downloaded
```

I think we should probably have links to more explanation for most of these as well.